nucleof072rb/Core/Src/main.c

```diff
@@ -144,6 +204,39 @@ void SystemClock_Config(void)
 
 /* USER CODE BEGIN 4 */
 
+double ADCToPWM(int ADC_val) {
+	return ((ADC_val / 1023) * 3000) + 3000;
```

Comment:
suggest find the relevant count values by using __HAL_TIM_GET_AUTORELOAD instead of hardcoding the number.
`( 0.05 + (adc_val * 0.05 / 1023) ) * __HAL_TIM_GET_AUTORELOAD(&htim1)`